gribscan/rotated_grid.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import division
+
+import numpy as np
+import eccodes as ec
+
+pi = np.pi
+rad = pi / 180.0
+radinv = 1.0 / rad
+
+
+def rot_to_reg(pole_lon, pole_lat, lon, lat):
+    """Rotates from rotated grid to regular grid
+
+    Parameters
+    ----------
+    pole_lon : float
+        Longitude of pole in degrees
+    pole_lat : float
+        Latitude of pole in degrees
+    lon : float, list
+        Longitudes of points in grid. Can also be a single point.
+    lat : float, list
+        Latitudes of points in grid. Can also be a single point.
+
+    Returns
+    -------
+    lon1 : float, list
+        Longitudes of regular grid. Same shape as input.
+    lat1 : float, list
+        Latitudes  of regular grid. Same shape as input.
+    """
+    sin_pole = np.sin(rad * (pole_lat + 90.0))
+    cos_pole = np.cos(rad * (pole_lat + 90.0))
+
+    sin_lon = np.sin(rad * lon)
+    cos_lon = np.cos(rad * lon)
+    sin_lat = np.sin(rad * lat)
+    cos_lat = np.cos(rad * lat)
+
+    lat_tmp = cos_pole * sin_lat + sin_pole * cos_lat * cos_lon
+
+    try:
+        lat_tmp[np.where(lat_tmp < -1.0)] = -1.0
+        lat_tmp[np.where(lat_tmp > 1.0)] = 1.0
+    except TypeError:
+        if lat_tmp < -1.0:
+            lat_tmp = -1.0
+        if lat_tmp > 1.0:
+            lat_tmp = 1.0
+
+    lat2 = np.arcsin(lat_tmp) * radinv
+
+    cos_lat2 = np.cos(lat2 * rad)
+
+    cos_tmp = (cos_pole * cos_lat * cos_lon - sin_pole * sin_lat) / cos_lat2
+
+    try:
+        cos_tmp[np.where(cos_tmp < -1.0)] = -1.0
+        cos_tmp[np.where(cos_tmp > 1.0)] = 1.0
+    except TypeError:
+        if cos_tmp < -1.0:
+            cos_tmp = -1.0
+        if cos_tmp > 1.0:
+            cos_tmp = 1.0
+
+    tmp_sin_lon = cos_lat * sin_lon / cos_lat2
+    tmp_cos_lon = np.arccos(cos_tmp) * radinv
+
+    try:
+        tmp_cos_lon[np.where(tmp_sin_lon < 0.0)] = -tmp_cos_lon[
+            np.where(tmp_sin_lon < 0.0)
+        ]
+    except IndexError:
+        if tmp_sin_lon < 0.0:
+            tmp_cos_lon = -tmp_cos_lon
+
+    lon2 = tmp_cos_lon + pole_lon
+
+    return lon2, lat2
+
+
+def get_gids(gribfile: str, TextIOWrapper: bool = False) -> list:
```

Comment:
This function only gets called by `get_latlons` which in turn isn't called. Thus this function should likely not be included as well.